Disambiguate shared and static libs `.lib` files on Windows using MSVC

adriendelsalle opened the pull request:
Description
---

Disambiguate shared and static libs `.lib` files on Windows using MSVC
Suffix library output name with `_static` when building both static and shared libs on Windows

Closes #454

mmuetzel commented on cmake/macros/SundialsAddLibrary.cmake:
```diff
@@ -325,10 +325,17 @@ macro(sundials_add_library target)
-          OUTPUT_NAME ${sundials_add_library_OUTPUT_NAME}
-          CLEAN_DIRECT_OUTPUT 1
-        )
+        if(WIN32 AND ${_libtype} MATCHES "STATIC")
```
Consider changing this to the following:
``` suggestion
        if((MSVC OR ("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")) AND ${_libtype} MATCHES "STATIC")
```
For MinGW, `WIN32` is true. But it doesn't have the naming conflict that compilers that target MSVC have.

adriendelsalle replied:
Thanks for your help! I understood from @balos1 that this PR should be closed in favor to a more generic/customizable approach.
Let me know!

balos1 commented:
Please add a note about this change to `CHANGELOG.md` and `doc/shared/RecentChanges.rst`

adriendelsalle commented:
@balos1 
- rebased on `develop`
- added changes to `CHANGELOG.md` and `doc/shared/RecentChanges.rst`
- fixed `SUNDIALS_EXPORT` macro incorrectly set when building both variants -> let me know if you prefer another PR for this, even if it's related to building both static and shared libs
